botcore/utils/scheduling.py

```diff
@@ -209,7 +209,7 @@ def _task_done_callback(self, task_id: typing.Hashable, done_task: asyncio.Task)
 
 
 def create_task(
-    coro: typing.Awaitable,
+    coro: typing.Coroutine,
```

Comment:
While we're here this can use a typevar for the corosđ return type to preserve the type information on the task it returns

Reply:
Yeah that's a good call. ecb9a7e4

Annoyingly, the docs don't seem to carry that return type through, and just leave it as `Task`, but at leat it will help static analysis tools.

Reply:
That happens because the type is documented in its `Returns:`... but trying it out when removed, sphinx-autodoc-typehints fails in resolving it as it sees it as `_asyncio.Task`, and even with that fixed we wouldn't get the return type because 3.9 doesn't seem to store the annotation on Task for some reason (or at least the 3.9 I have as I can't find anything about it in the changelogs)

For now we can do `asyncio.Task[TASK_RETURN]` in the docstring